packages/api/src/interactors/UpdateServiceAnswerInteractor.ts

```diff
@@ -0,0 +1,80 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import {
+	ServiceAnswerInput,
+	ServiceResponse,
+	StatusType
+} from '@cbosuite/schema/dist/provider-types'
+import { Localization } from '~components'
+import { DbServiceAnswer, ServiceCollection } from '~db'
+import { createDBServiceAnswer, createGQLService } from '~dto'
+import { Interactor } from '~types'
+
+export class UpdateServiceAnswerInteractor
+	implements Interactor<ServiceAnswerInput, ServiceResponse>
+{
+	#localization: Localization
+	#services: ServiceCollection
+
+	public constructor(localization: Localization, services: ServiceCollection) {
+		this.#localization = localization
+		this.#services = services
+	}
+
+	public async execute(serviceAnswer: ServiceAnswerInput): Promise<ServiceResponse> {
+		if (!serviceAnswer.serviceId) {
+			return {
+				service: null,
```

Comment:
We repeat a same-ish block here a few times, let's add a `failureWithMessage(message: string)` function at the module level that assembles this object

Reply:
will apply in the next iteration